File: models/Role.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}dateCreated`,
                label: `Timestamp when the role assignment was created (as [year, month, day, hour, minute, second, nanosecond]) - [${labelPrefix}dateCreated]`,
                list: true,
                type: 'integer',
            },
            {
                key: `${keyPrefix}dateModified`,
                label: `Timestamp when the role assignment was last modified - [${labelPrefix}dateModified]`,
                list: true,
                type: 'integer',
            },
            {
                key: `${keyPrefix}roleId`,
                label: `Unique role ID (UUID) - [${labelPrefix}roleId]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}communityId`,
                label: `Community ID or level associated with the role - [${labelPrefix}communityId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}name`,
                label: `Name of the role - [${labelPrefix}name]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}description`,
                label: `Description of the role - [${labelPrefix}description]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}level`,
                label: `Role level or hierarchy - [${labelPrefix}level]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}type`,
                label: `Role type/category - [${labelPrefix}type]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}inactive`,
                label: `Whether the role is inactive - [${labelPrefix}inactive]`,
                type: 'boolean',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'dateCreated': bundle.inputData?.[`${keyPrefix}dateCreated`],
            'dateModified': bundle.inputData?.[`${keyPrefix}dateModified`],
            'roleId': bundle.inputData?.[`${keyPrefix}roleId`],
            'communityId': bundle.inputData?.[`${keyPrefix}communityId`],
            'name': bundle.inputData?.[`${keyPrefix}name`],
            'description': bundle.inputData?.[`${keyPrefix}description`],
            'level': bundle.inputData?.[`${keyPrefix}level`],
            'type': bundle.inputData?.[`${keyPrefix}type`],
            'inactive': bundle.inputData?.[`${keyPrefix}inactive`],
        }
    },
}

File: client/scala-gatling/src/gatling/scala/org/openapitools/client/model/Role.scala

package org.openapitools.client.model


case class Role (
    /* Timestamp when the role assignment was created (as [year, month, day, hour, minute, second, nanosecond]) */
    _dateCreated: Option[List[Integer]],
    /* Timestamp when the role assignment was last modified */
    _dateModified: Option[List[Integer]],
    /* Unique role ID (UUID) */
    _roleId: Option[String],
    /* Community ID or level associated with the role */
    _communityId: Option[Integer],
    /* Name of the role */
    _name: Option[String],
    /* Description of the role */
    _description: Option[String],
    /* Role level or hierarchy */
    _level: Option[Integer],
    /* Role type/category */
    _type: Option[String],
    /* Whether the role is inactive */
    _inactive: Option[Boolean]
)
object Role {
    def toStringBody(var_dateCreated: Object, var_dateModified: Object, var_roleId: Object, var_communityId: Object, var_name: Object, var_description: Object, var_level: Object, var_type: Object, var_inactive: Object) =
        s"""
        | {
        | "dateCreated":$var_dateCreated,"dateModified":$var_dateModified,"roleId":$var_roleId,"communityId":$var_communityId,"name":$var_name,"description":$var_description,"level":$var_level,"type":$var_type,"inactive":$var_inactive
        | }
        """.stripMargin
}
